src/rest-api/class-base-api-controller.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/**
+ * Base API Controller
+ *
+ * @package Parsely
+ * @since   3.17.0
+ */
+
+declare(strict_types=1);
+
+namespace Parsely\REST_API;
+
+use Parsely\Parsely;
+
+/**
+ * Base API Controller.
+ *
+ * Used to define the namespace, version, and endpoints for an API controller. API controllers
+ * should extend this class and implement the `init` method to register endpoints.
+ *
+ * @since 3.17.0
+ */
+abstract class Base_API_Controller {
+
+	/**
+	 * The API namespace.
+	 *
+	 * @var string|false
+	 */
+	public const NAMESPACE = false;
+
+	/**
+	 * The API version.
+	 *
+	 * @var string
+	 */
+	public const VERSION = '';
```

Comment:
Just noting that an alternative approach to defining constants that we want child classes to mandatorily set, is to write getter/setter functions, turning the const into a protected variable, and making the setter an abstract method that children are forced to implement.

While this approach is a bit more verbose, it enforces class expectations and blocks us in the IDE level, before any code runs. This means it should also save us from any error/exception handling down the road, which would cause issues if we forgot to include it.

I'm not saying we should absolutely go that route, but it might be worth considering in this redesign.

Reply:
Yeah, I like that idea. I did a small rework on that part to include abstract getters for the namespace and endpoint name, and a couple getters for the version and route prefix (that return empty by default).

Let me know what you think of that approach.

Reply:
I've looked at the controllers in your commit and I think I prefer the new approach. Let me know if you disagree though. I saw that `version` and `route` aren't abstract, but we can  abstract them if need be.